src/rand.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Authors (alpha order): Adrian Catangiu @acatangiu, Alexandra Lordache @aghecenco, Alexandru Agache @alexandruag, Diana Popa @dianpopa, Matt Forrester @forbesmyester, Radu Weiss @raduweiss
+
+extern crate libc;
+use std::ffi::OsString;
+use std::str;
+
+pub fn timestamp_cycles() -> u64 {
+    #[cfg(target_arch = "x86_64")]
+    // Safe because there's nothing that can go wrong with this call.
+    unsafe {
+        std::arch::x86_64::_rdtsc() as u64
+    }
+    #[cfg(not(target_arch = "x86_64"))]
+    {
+        let mut ts = libc::timespec {
+            tv_sec: 0,
+            tv_nsec: 0,
+        };
+
+        unsafe {
+            libc::clock_gettime(libc::CLOCK_MONOTONIC, &mut ts);
+        }
+        (ts.tv_sec as u64) * 1000000000 + (ts.tv_nsec as u64)
+    }
+}
+
+fn timespec_to_us(time_struct: &libc::timespec) -> u64 {
+    (time_struct.tv_sec as u64) * 1_000_000 + (time_struct.tv_nsec as u64) / 1000
```

Comment:
This code could benefit from some comments about what is the logic here.
Can you also add defines for the magic numbers or explain in a comment what they represent?

Reply:
Well this is a straight port of `firecracker/fc_util.rs` and I don't really understand it... However the function is not required so removed.

Reply:
git-blaming @aghecenco. Can you helps us with some comments on this function please?

Reply:
The function was removed from this PR, and also from a PR opened in Firecracker - https://github.com/firecracker-microvm/firecracker/pull/1174 - so there's not much use in documenting it, I think